### ISSUE TITLE
Not enough instructions to compile and run

### ISSUE THREAD

**Issue by Sundar69** [YuvalNirkin/face_swap#29]
Please provide more instructions to compile. Which cmake file to use it to compile. The results look good but the steps to compile is absent. The issue is for people who want to build and see the example first before learning ml. Thats how some type of people learn ml.

**Comment by YuvalNirkin** [comment]
You're absolutely right, I will update the instructions.

**Comment by lawiz22** [comment]
And you should say that it take at least a 12 Go GPU to use this deep model. BTW Really good paper , i'm gonna take a second read on it tonight. Would love to test the code but i face the hard reality when it came to face_segmentation

**Comment by YuvalNirkin** [comment]
Thank you.

For now I released a [new FCN model](https://github.com/YuvalNirkin/face_segmentation/releases/download/1.1/face_seg_fcn8s_300_no_aug.zip) for lower resolution images which is more memory efficient.
I'll update the code soon for this new model.

**Comment by lawiz22** [comment]
![figure_1](https://user-images.githubusercontent.com/7004883/42549235-2262d50c-8499-11e8-9d91-472991c50154.png)
It's working for me thank you

**Comment by ej90** [comment]
@YuvalNirkin This is fascinating work, I appreciate you putting this up, but like @Sundar69 am having trouble building and testing the tool. Any chance of putting up a Docker container up?

**Comment by YuvalNirkin** [comment]
@ej90 @Sundar69 Thank you.  I am not sure about a docker container as I am kind of busy right now but I will add a detailed installation guide for Linux. I am also looking into removing some of the dependencies and adding Python binding to make this code more accessible.

@lawiz22 I also added a post-processing option that will make the segmentation smoother,  especially useful for the lower resolution model.

**Comment by ej90** [comment]
@YuvalNirkin Wonderful, thanks for the response! Really looking forward to it. Again, thanks for posting all of this -- really cool stuff.

**Comment by lawiz22** [comment]
Thank @YuvalNirkin  i'm looking foward to it!  @ej90  : here's a docker version : https://github.com/RudyChin/face_swap_docker

**Comment by YuvalNirkin** [comment]
@Sundar69 @ej90 @lawiz22 Sorry for the late response, I have updated the code to make it easier to build. I removed some of the dependencies, most importantly I replaced OpenGL with a software renderer so the code should build on gui-less servers without any problems.

As requested I released a detailed [Ubuntu installation guide](https://github.com/YuvalNirkin/face_swap/wiki/Ubuntu-Installation-Guide), I hope it will make it easier to install this repository.
